tests/Aspire.Hosting.Tests/Utils/LoggerNotificationExtensions.cs

```diff
@@ -65,9 +62,9 @@ private static async Task WatchNotifications(DistributedApplication app, IResour
 
         try
         {
-            await foreach (var resourceEvent in resourceNotificationService.WatchAsync(cancellationToken).ConfigureAwait(false))
+            await foreach (var resourceEvent in resourceNotificationService.WatchAsync(cancellationTokenSource.Token).ConfigureAwait(false))
```

Comment:
Just curious - why do we continue watching for resource notifications once we have found one matching `resourceName`?

Would these resource events be like - starting up, running, stopped etc? What happens if a resource starts, we call `WatchResourceLogs`, then it crashes, and dcp starts it up again? Do we need to call `WatchResourceLogs` again?
Does the `resourceEvent.ResourceId` stay the same over multiple notification messages, and even with restarts?

Reply:
There could actually be multiple "resourceId" for a single resource. Example when we look for "oracle" there is another one like "oracle-xxxx-yyyy" for the docker container. And they will both match the resource name.